protocol/assertion.go

```diff
@@ -132,7 +137,16 @@ func (p *ParsedCredentialAssertionData) Verify(storedChallenge string, relyingPa
 
 	sigData := append(p.Raw.AssertionResponse.AuthenticatorData, clientDataHash[:]...)
 
-	key, err := webauthncose.ParsePublicKey(credentialBytes)
+	var (
+		key interface{}
+		err error
+	)
+
+	if appID == "" {
```

Comment:
```suggestion
	if appID == "" || attestationType != "fido-u2f" {
```

Maybe we should pass the attestation type from the loginCredential here as well for safety?

Reply:
See this commit for an example, plus validation that the appid was actually set by the server by recording it in the session data:

https://github.com/authelia/webauthn/pull/7/commits/1f0ad2d885b2f8e3589a9ad7b0a542d754aefde8